### PR TITLE
Update remove member logic to trigger rerender

### LINKED ISSUE #165: [PE-D][Tester D] UI does not update task assignment after members are deleted from a team
For example in the screenshot, task 2 is still assigned to Bernice Yu even though they have been deleted from the team. 

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: pewggls/ped#7

### PR DESCRIPTION
**Pull request by mjgui**
Fixes #165

### CODE REVIEW

File: src/main/java/seedu/address/model/team/Task.java [hunk 1]
```diff
@@ -166,23 +183,15 @@ public boolean checkAssignee(Person assignee) {
...
      * Set a new deadline and returns a new Task with that deadline
      */
-    public Task setDeadline(LocalDateTime date) {
-        String name = getName();
-        List<Person> assignees = getAssigneesList();
-        boolean completionStatus = isComplete();
-        LocalDateTime newDate = date;
-        return new Task(name, assignees, completionStatus, newDate);
+    public Task setDeadline(LocalDateTime newDeadline) {
+        return new Task(name, assignees, completionStatus, newDeadline);
```
Comment (marcuspang) [inline src/main/java/seedu/address/model/team/Task.java:187]: Wait, why did you remove the getter functions? Feels like this change isn't good practice since the getter functions may have other implementation details abstracted away

File: src/main/java/seedu/address/model/team/Task.java [hunk 2]
```diff
@@ -49,16 +49,37 @@ public class Task {
...
 
+    /**
+     * Constructs a {@code Task}.
+     *
+     * @param name A valid task name.
+     * @param assignees A UniquePersonList of persons this task is assigned to
+     * @param completionStatus Boolean representing whether the task is completed
+     * @param deadline Datetime representing deadline of the task
+     */
+    public Task(String name, UniquePersonList assignees, boolean completionStatus, LocalDateTime deadline) {
```
Comment (marcuspang) [inline src/main/java/seedu/address/model/team/Task.java:74]: I don't see a point for this constructor since we shouldn't be passing the same `assignees` reference in the other getter functions

**Review by marcuspang — CHANGES_REQUESTED** [review]
I feel some changes made were unnecessary, but the rest looks fine

File: src/main/java/seedu/address/model/team/Task.java (hunk 2 above)
Comment (mjgui) [reply src/main/java/seedu/address/model/team/Task.java:74]: this constructor is meant to allow for reusing of a `UniquePersonList` reference when it is known to be no longer used

to prevent misuse, i've made it private

File: src/main/java/seedu/address/model/team/Task.java (hunk 1 above)
Comment (mjgui) [reply src/main/java/seedu/address/model/team/Task.java:187]: in this case `setDeadline` should return a copy of the internal variables (and thus should not include any logic)
if we need any logic to be done before constructing a copy it should be done inside the constructor, not in the getter

File: src/main/java/seedu/address/model/team/TaskList.java [hunk 3]
```diff
@@ -174,8 +174,13 @@ public String toString() {
...
      */
     public void removeAssigneeIfExists(Person person) {
-        for (Task task : internalList) {
-            task.removeAssigneeIfExists(person);
+        for (int i = 0; i < internalList.size(); i++) {
+            Task task = internalList.get(i);
+            if (task.checkAssignee(person)) {
+                Task newTask = task.removeAssignee(person);
+                internalList.set(i, newTask);
+                return;
```
Comment (marcuspang) [inline src/main/java/seedu/address/model/team/TaskList.java:182]: Just checking, is this meant to be here

**Review by marcuspang — APPROVED** [review]
Not sure if the early return is intentional here, but the rest LGTM

File: src/main/java/seedu/address/model/team/TaskList.java (hunk 3 above)
Comment (mjgui) [reply src/main/java/seedu/address/model/team/TaskList.java:182]: eh crap it's not, thanks for the catch
